### PR TITLE
Niloofar / Footer's server time implementation 

### PR DESCRIPTION
**Pull request by niloofar-deriv**
This PR includes all of the file changes of footer's server time!

Screenshot 2023-11-21 at 12 46 35 PM

### CODE REVIEW

File: src/components/layout/footer/NetworkStatus.tsx [hunk 1]
```diff
@@ -3,9 +3,17 @@ import Tooltip from 'Components/common/tooltip';
 import useNetworkStatus from 'Hooks/useNetworkStatus';
 
 const NetworkStatus = () => {
```
Comment (niloofar-deriv) [inline src/components/layout/footer/NetworkStatus.tsx:12]: Just wanted to improve useNetworkStatus hook

File: src/constants/moment.ts
Comment (shayan-deriv) [inline src/constants/moment.ts:1]: these are utilities and should not be under constants directory.

File: src/constants/moment.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import moment from 'moment';
+
+export const epochToGMTFormat = (time: number) => moment.unix(time).utc().format('YYYY-MM-DD HH:mm:ss [GMT]');
```
Comment (shayan-deriv) [inline src/constants/moment.ts]:
```suggestion
export const epochToUtc = (time: number, format: string, timezone: string) => moment.unix(time).utc().format(`${format} [${timzone}]`);
```
let's make it more general.

File: src/constants/moment.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+export const epochToLocalFormat = (time: number) => moment.unix(time).utc().local().format('YYYY-MM-DD HH:mm:ss Z');
```
Comment (shayan-deriv) [inline src/constants/moment.ts]:
```suggestion
export const epochToLocal = (time: number, format: string, timezone: string) => moment.unix(time).utc().local().format(`${format} ${timezone}`);
```

File: src/hooks/useServerTime.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        if (data) {
+            setServerTime(data.time ?? currentDate);
+
+            timeInterval = setInterval(() => {
+                setServerTime(prev => prev + 1);
+            }, 1000);
+        }
+
+        return () => clearInterval(timeInterval);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (shayan-deriv) [inline src/hooks/useServerTime.ts:21]: why not just updating the dependency array?

**Review by shayan-deriv — COMMENTED** [review]
regarding the tests , let's improve them in a separate call. we need more test cases

File: src/hooks/useServerTime.ts (hunk 4 above)
Comment (niloofar-deriv) [reply src/hooks/useServerTime.ts:21]: If I put that inside the dependency array it will throw an error and the timer won't work that's why I did that

**Review by shayan-deriv — APPROVED** [review]

**Review by ako-deriv — APPROVED** [review]

**Review by balakrishna-deriv — APPROVED** [review]

**Pull request merged**